cranelift/codegen/src/isa/x64/lower.isle

```diff
@@ -3510,6 +3510,21 @@
 
 ;; Rules for `shuffle` ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 
+;; Special case for the `punpckhbw` instruction which interleaves the upper
+;; lanes of the two input registers.
+(rule 5 (lower (shuffle a b (vec_mask_from_immediate mask)))
+      (if-let $true (vec_mask_is mask 8 24 9 25 10 26 11 27 12 28 13 29 14 30 15 31))
```

Comment:
Seems reasonable to me!

Reply:
And now that I'm looking at that revision, it occurs to me that if you make a `u128_from_vec_mask` extractor instead of `vec_mask_is`, you can put these two rules at the same priority because ISLE will be able to tell that their different masks don't overlap. That should generate better pattern-matching code. Not super important, but would be nice.